lib/Sema/TypeCheckType.cpp

```diff
@@ -188,6 +187,25 @@ Type TypeResolution::resolveDependentMemberType(
 
   auto *concrete = ref->getBoundDecl();
 
+  if (auto concreteBase = genericSig->getConcreteType(baseTy)) {
+    bool hasUnboundOpener = !!getUnboundTypeOpener();
```

Comment:
Should this be `!`?

Reply:
Nope, I want the value of `getUnboundTypeOpener` as a `bool`. This `!!` pattern is used elsewhere in TypeCheckType.cpp on `getUnboundTypeOpener` to do the same thing.

Reply:
Oh interesting! I didn’t know we had a `!!`.

Reply:
I didn't either until I looked at the other call-sites of `TypeChecker::isUnsupportedMemberTypeAccess` 😄